Update test_post_unauthorized API test

elyezer opened the pull request:
Update the entities checked on the test to include just entities which
are possible to post on its path.

```python
nosetests tests/foreman/api/test_multiple_paths.py -m test_post_unauthorized
.
----------------------------------------------------------------------
Ran 1 test in 36.721s

OK
```

Closes #3020

omaciel commented:
ACK

sthirugn commented on tests/foreman/api/test_multiple_paths.py:
```diff
@@ -236,7 +236,10 @@ def test_post_unauthorized(self):
+        post_entities = list(valid_entities())
+        # TemplateKind is a valid entity but it is read-only.
+        post_entities.remove(entities.TemplateKind)
+        for entity_cls in post_entities:
```
Can we use the same approach as in https://github.com/SatelliteQE/robottelo/pull/3029/files#diff-4d5fa73ca5b6eca8886d32b3df8fc316R265?  This will ease adding/removing entities with much code logic change.

oshtaier replied:
+1

oshtaier commented:
ACK if mentioned behavior is intended

sthirugn commented:
Please hold on with merging, I have a comment

omaciel commented:
@oshtaier having trouble to sleep? :)

sthirugn commented:
dreaming of robottelo :)

oshtaier commented:
@omaciel just decided to spend 5 mins after dinner for my love project :)

omaciel commented:
![image](http://www.reactiongifs.com/r/strtldwk.gif)

elyezer commented:
Test continue passing after addressing the comments.

```
$ nosetests tests/foreman/api/test_multiple_paths.py -m test_post_unauthorized
.
----------------------------------------------------------------------
Ran 1 test in 41.846s

OK
```

omaciel commented:
ACK
